src/components/Main.tsx

```diff
@@ -0,0 +1,687 @@
+        }}
+      />
+      <Stack.Screen
+        name="currencySettings"
+        component={ifLoggedIn(CurrencySettingsScene)}
+        options={{
+          headerTitle: props => <CurrencySettingsTitle />,
+          headerRight: () => null
+        }}
+      />
+      <Stack.Screen
+        name="promotionSettings"
+        component={ifLoggedIn(PromotionSettingsScene)}
+        options={{
+          title: s.strings.title_promotion_settings,
+          headerRight: () => null
+        }}
+      />
+      <Stack.Screen
+        name="defaultFiatSetting"
+        component={ifLoggedIn(DefaultFiatSettingScene)}
+        options={{
+          headerRight: () => null
+        }}
+      />
+      <Stack.Screen
+        name="notificationSettings"
+        component={ifLoggedIn(NotificationScene)}
+        options={{
+          title: s.strings.settings_notifications,
+          headerRight: () => null
+        }}
+      />
+      <Stack.Screen
+        name="currencyNotificationSettings"
+        component={ifLoggedIn(CurrencyNotificationScene)}
+        options={{
+          headerTitle: props => <CurrencySettingsTitle />,
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="pluginView"
+        component={ifLoggedIn(GuiPluginViewScene)}
+        options={{
+          headerTitle: () => <ParamHeaderTitle<'pluginView'> fromParams={params => params.plugin.displayName} />,
+          headerRight: () => <HeaderTextButton type="exit" placement="right" />,
+          headerLeft: () => <PluginBackButton />
+        }}
+      />
+
+      <Stack.Screen
+        name="termsOfService"
+        component={ifLoggedIn(TermsOfServiceComponent)}
+        options={{
+          title: s.strings.title_terms_of_service
+        }}
+      />
+
+      <Stack.Screen name="fioAddressList" component={ifLoggedIn(FioAddressListScene)} />
+
+      <Stack.Screen
+        name="fioAddressRegister"
+        component={ifLoggedIn(FioAddressRegisterScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressRegisterSelectWallet"
+        component={ifLoggedIn(FioAddressRegisterSelectWalletScene)}
+        options={{
+          title: s.strings.title_fio_address_confirmation,
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioDomainRegister"
+        component={ifLoggedIn(FioDomainRegisterScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+      <Stack.Screen
+        name="fioDomainRegisterSelectWallet"
+        component={ifLoggedIn(FioDomainRegisterSelectWalletScene)}
+        options={{
+          title: s.strings.title_register_fio_domain,
+          headerRight: () => null
+        }}
+      />
+      <Stack.Screen
+        name="fioDomainConfirm"
+        // @ts-expect-error
+        component={ifLoggedIn(FioNameConfirmScene)}
```

Comment:
Visit the FioNameConfirmScene, and change the props to take `'fioNameConfirm' | "fioDomainConfirm"` to get rid of this.